lib/membrane_h264_ffmpeg/parser.ex

```diff
@@ -391,12 +404,24 @@ defmodule Membrane.H264.FFmpeg.Parser do
     }
   end
 
-  defp carries_parameters_in_band?(<<payload::binary-size(40), _rest::binary>>) do
+  defp carries_parameters_in_band?(payload) do
     types =
       NALu.parse(payload)
       |> elem(0)
       |> Enum.map(& &1.metadata.h264.type)
 
-    Enum.all?(@required_parameter_nalu, &Enum.member?(types, &1))
+    {parameters, data} = Enum.split_while(types, &Enum.member?(@nalu_allowed_before_data, &1))
```

Comment:
1. I'd rename that to something more explanatory: `parameter_nalus` and `data_nalus`
2. Instead of `Enum.member?` I'd rather use `MapSet.member?`